sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/connectionstring/ConnectionSettings.java

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.implementation.connectionstring;
+
+import com.azure.core.util.logging.ClientLogger;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A dictionary representation of all settings in a connection string.
+ */
+final class ConnectionSettings implements Cloneable {
+    private final Map<String, String> settings;
+
+    /**
+     * Checks a given settings exists.
+     *
+     * @param name the setting name
+     * @return true if setting exists, false otherwise
+     */
+    public boolean hasSetting(String name) {
+        return this.settings.containsKey(name);
+    }
+
+    /**
+     * Remove a setting with given name if exists.
+     *
+     * @param name the setting name
+     */
+    public void removeSetting(String name) {
+        this.settings.remove(name);
+    }
+
+    /**
+     * Get value of the setting with given name.
+     *
+     * @param name the setting name
+     * @return the setting value if exists
+     */
+    public String getSettingValue(String name) {
+        return this.settings.get(name);
+    }
+
+    /**
+     * @return true if there no settings exists, false otherwise.
+     */
+    public boolean isEmpty() {
+        return this.settings.isEmpty();
+    }
+
+    /**
+     * Set a setting.
+     *
+     * @param name the setting name
+     * @param value the setting value
+     */
+    public void setSetting(String name, String value) {
+        this.settings.put(name, value);
+    }
+
+    /**
+     * Creates {@link ConnectionSettings} from the given connection string.
+     *
+     * @param connString the connection string
+     * @param logger the logger
+     * @return the ConnectionSettings
+     */
+    public static ConnectionSettings fromConnectionString(final String connString,
+                                                          final ClientLogger logger) {
+        HashMap<String, String> map = new HashMap<>();
+        final String[] settings = connString.split(";");
+        for (int i = 0; i < settings.length; i++) {
+            String setting = settings[i].trim();
+            if (setting.length() > 0) {
+                final int idx = setting.indexOf("=");
+                if (idx == -1
+                        || idx == 0
+                        || idx == settings[i].length() - 1) {
+                    // handle no_equal_symbol, "=Bar", "Foo="
+                    throw logger.logExceptionAsError(new IllegalArgumentException("Invalid connection string."));
```

Comment:
Is it actually invalid for a connection string piece to have no value?

Reply:
we can have either strict check or relaxed check on absence of value, here we're being bit strict, if needed it's easy to relax it later than the other way. 

The rule at least now is: If user want a setting then specify it as a valid pair of non-null key-value, if setting is not needed then don't include it in the connection string.